pkg/driver/trace_driver.go

```diff
@@ -100,6 +100,27 @@ func (d *Driver) runCSVWriter(records chan interface{}, filename string, writerD
 	writerDone.Done()
 }
 
+func DAGCreation(functions []*common.Function) (*common.Node, int64) {
+	var head *common.Node
+	var prev *common.Node
+	var numberOfNodes int64
+	// Assigning nodes one after another
+	for _, function := range functions {
+		node := &common.Node{
+			Function: function,
+			NextNode: nil,
+		}
+		if head == nil {
+			head = node
+		} else {
+			prev.NextNode = node
+		}
+		prev = node
+		numberOfNodes++
+	}
+	return head, numberOfNodes
```

Comment:
numberOfNodes that is returned is just size of the functions array. I don't see a reason to count them again.